tracer/src/Datadog.Trace/Ci/CIVisibility.cs

```diff
@@ -418,5 +440,39 @@ private static async Task GetIntelligentTestRunnerSkippableTestsAsync()
                 Log.Error(ex, "ITR: Error getting skippeable tests.");
             }
         }
+
+        private static bool IsEventPlatformProxySupportedByAgent(ref IDiscoveryService discoveryService)
+        {
+            var eventPlatformProxyEnabled = false;
+            discoveryService = DiscoveryService.Create(new ImmutableExporterSettings(_settings.TracerSettings.Exporter));
+            AgentConfiguration? agentConfiguration = null;
+            ManualResetEventSlim manualResetEventSlim = new(false);
+            LifetimeManager.Instance.AddShutdownTask(() => manualResetEventSlim.Set());
+
+            Log.Debug("Waiting for agent configuration...");
+            var discoveryServiceCopy = discoveryService;
+            discoveryServiceCopy.SubscribeToChanges(CallBack);
+            void CallBack(AgentConfiguration aConfiguration)
+            {
+                agentConfiguration = aConfiguration;
+                manualResetEventSlim.Set();
+                discoveryServiceCopy.RemoveSubscription(CallBack);
+            }
+
+            // We wait up to 5 seconds for the configuration to be retrieved.
+            manualResetEventSlim.Wait(5_000);
+            if (!string.IsNullOrEmpty(agentConfiguration?.EventPlatformProxyEndpoint))
```

Comment:
_maybe_ want a `Volatile.Read(ref agentConfiguration)`? As it could be set on a different thread, there's no async here, and you don't know if the event was set or not...

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/3322/commits/e65ccbfc53a8a19b691e32b998542e4a8b174ba0